cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -981,23 +981,28 @@ func listIndexed(indexDir string) []uint32 {
 //
 // If main is true we will delete older temp directories left around. main is
 // false when this is a debug command.
-func setupTmpDir(main bool, index string) error {
-	// change the target tmp directory depending on if its our main daemon or a
+func setupTmpDir(logger sglog.Logger, main bool, index string) error {
+	// change the target tmp directory depending on if it's our main daemon or a
 	// debug sub command.
 	dir := ".indexserver.debug.tmp"
 	if main {
 		dir = ".indexserver.tmp"
 	}
 
 	tmpRoot := filepath.Join(index, dir)
-	if err := os.MkdirAll(tmpRoot, 0755); err != nil {
-		return err
+
+	if main {
+		err := os.RemoveAll(tmpRoot)
+		if err != nil {
+			logger.Error("failed to remove tmp dir", sglog.String("tmpRoot", tmpRoot), sglog.Error(err))
+		}
 	}
-	if !tmpfriend.IsTmpFriendDir(tmpRoot) {
-		_, err := tmpfriend.RootTempDir(tmpRoot)
+
+	if err := os.MkdirAll(tmpRoot, 0755); err != nil {
 		return err
 	}
-	return nil
+
+	return os.Setenv("TMPDIR", tmpRoot)
```

Comment:
This is a new change, right? Are we sure this doesn't have any unintended side-effects?

Reply:
We previously did this in `tmpfriend.RooTempDir(tmpRoot)` on line 997

Reply:
Got it, thanks!